### PR TITLE
feat: event pages

### LINKED ISSUE #279: Create event pages
Please create event pages similar to open event.

Screenshot from 2022-02-10 12-47-58

### LINKED ISSUE #308: feat: License
Currently `open-event-next` doesn't have any License. We need a License for the open-source project and npm/yarn is also complaining about this as the License isn't present in the `package.json` file

### PR DESCRIPTION
**Pull request by niloysikdar**
Fixes #279 
Fixes #308

### CODE REVIEW

**Comment by niloysikdar** [comment]
Deploy will fail @mariobehling , I need to pull the changes from #278 after merging that PR

**Comment by niloysikdar** [comment]
Fixes #308

**Comment by niloysikdar** [comment]
The CI is failing while executing `npx next export`
Check [here](https://github.com/fossasia/open-event-next/runs/5337297144?check_suite_focus=true)

It is failing due to the `Image` component which I am importing from `next/image`
Here is the issue: [link](https://nextjs.org/docs/messages/export-image-api)
Also one useful thread: [link](https://stackoverflow.com/questions/61724368/what-is-the-difference-between-next-export-and-next-build-in-next-js)

What can we do now @progmatic-99 @pc-beast ?

File: package.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "eslint": "^8.9.0",
     "eslint-config-prettier": "^8.4.0",
     "eslint-plugin-jsx-a11y": "^6.5.1",
-    "eslint-plugin-react": "^7.28.0",
+    "eslint-plugin-react": "7.28.0",
```
Comment (niloysikdar) [inline package.json:63]: Due to [this issue](https://github.com/yannickcr/eslint-plugin-react/issues/3215), I was getting 300+ linting errors which caused of not allowing me to commit anything. That's why I fixed this one for now.

File: components/EventPage/JoinEvent.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+const ColorButton = styled(Button)(({ theme }) => ({
+  color: theme.palette.getContrastText(grey[300]),
+  backgroundColor: grey[300],
+  '&:hover': {
+    backgroundColor: grey[400],
+  },
+}))
+
+const LinkButton = styled(Button)(() => ({
```
[the remark this answers is not in the document]
Comment (niloysikdar) [reply components/EventPage/JoinEvent.tsx:23]: Created custom styled components cz didn't want to copy-paste same styles again and again

**Comment by niloysikdar** [comment]
Capture2
<br/>
Screenshot (188)
<br/>

<div>


<div/>

<br/>

<div>


<div/>



@mariobehling @progmatic-99 @pc-beast

**Comment by niloysikdar** [comment]
CI is failing, reason [here](https://github.com/fossasia/open-event-next/pull/309#issuecomment-1051075495)
@mariobehling @progmatic-99 @pc-beast

File: components/EventPage/JoinEvent.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  [`& .${tooltipClasses.tooltip}`]: {
+    backgroundColor: '#fff',
+    color: 'rgba(0, 0, 0, 0.87)',
+    maxWidth: 220,
+    fontSize: theme.typography.pxToRem(12),
+    border: '2px solid #dadde9',
+  },
+}))
+
+const CustomLink = styled('a')(({ theme }) => ({
```
Comment (niloysikdar) [inline components/EventPage/JoinEvent.tsx:27]: Didn't remove this component as we are using this same styling in multiple positions.
Copy-pasting the same styles again and again to make the codebase larger isn't the ideal solution, if we can refactor the code with some components, then I think we should proceed on that way only.

Though I have removed a few components and added the styles directly using props, those styles weren't repeating, but here the case is different.

**Review by niloysikdar — COMMENTED** [review]
Can we merge the pr if everything seems fine @mariobehling  @progmatic-99 

I am solving lots of conflicts every few days as the pr is getting older and older
Coming to the data fetching part, I am doing this thing in a follow-up pr, but if this pr will remain open, then again I need to solve large conflicts.

File: components/EventPage/JoinEvent.tsx (hunk 3 above)
Comment (progmatic-99) [reply components/EventPage/JoinEvent.tsx:27]: Maybe create a HOC or come up with some solution.

File: components/EventPage/TicketTable.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        <Button variant="contained">Order Now</Button>
+        <Button variant="contained" startIcon={<InfoIcon />} color="success">
+          Refund Policy
+        </Button>
+      </Box>
+    </Paper>
+  )
+}
+
+const TicketPart = ({
```
Comment (progmatic-99) [inline components/EventPage/TicketTable.tsx:52]: create an interface separately.

File: package.json [hunk 5]
```diff
@@ -62,7 +65,7 @@
     "eslint": "^8.10.0",
     "eslint-config-prettier": "^8.4.0",
     "eslint-plugin-jsx-a11y": "^6.5.1",
-    "eslint-plugin-react": "^7.29.2",
+    "eslint-plugin-react": "7.28.0",
```
Comment (progmatic-99) [inline package.json:68]: Why are you downgrading the already updated package??

File: pages/index.tsx [hunk 6]
```diff
@@ -48,7 +48,7 @@ export default function Index(props: Props): JSX.Element {
               The open source event solution for virtual and in-person events.
             </Trans>
           </Typography>
-          <Button variant="contained">Create Event</Button>
+          <Button variant="contained">CREATE EVENT</Button>
```
Comment (progmatic-99) [inline pages/index.tsx:51]: Why? It's already capitalized on the site.

Pls don't make unnecessary changes.

**Review by progmatic-99 — CHANGES_REQUESTED** [review]
CI test is failing. Please make it pass.
Don't push your vscode settings folder.

Thanks

File: components/EventPage/JoinEvent.tsx (hunk 3 above)
Comment (niloysikdar) [reply components/EventPage/JoinEvent.tsx:27]: Yes, sure, extracted the component and the logic

File: components/EventPage/TicketTable.tsx (hunk 4 above)
Comment (niloysikdar) [reply components/EventPage/TicketTable.tsx:52]: Okay, also extracted to a different component file with the proper interface

File: package.json (hunk 5 above)
Comment (niloysikdar) [reply package.json:68]: Already mentioned the reason [here](https://github.com/fossasia/open-event-next/pull/309#discussion_r814985920)

Also, I tried to use `"eslint-plugin-react": "^7.29.2"` , but getting this warning. We need to downgrade the TypeScript version in this case.
<be />
Capture

File: pages/index.tsx (hunk 6 above)
Comment (niloysikdar) [reply pages/index.tsx:51]: By default, MUI button has the default property of `textTransform: "uppercase"` for the Button component
Now I figured out that, almost 90+ % of buttons in the open-event-frontend isn't using uppercase

To prevent using `"textTransform: "none"` again and again for this majority of cases, I configured the theme as `textTransform: 'none'` for Button  [here](https://github.com/fossasia/open-event-next/blob/cded71650c085a2fe39a16972f70378f1aa04d04/src/theme.ts#L22)

That's why I changed this text

File: package.json (hunk 5 above)
Comment (niloysikdar) [reply package.json:68]: Already mentioned the reason [here](https://github.com/fossasia/open-event-next/pull/309#discussion_r814985920)

nvm, updating it.

**Comment by niloysikdar** [comment]
> CI test is failing. Please make it pass. Don't push your vscode settings folder.
> 
> Thanks

- I have already mentioned the CI failing issue [here](https://github.com/fossasia/open-event-next/pull/309#issuecomment-1051075495)
Again mentioned the same issue [here](https://github.com/fossasia/open-event-next/pull/309#issuecomment-1055268683)

- If you kindly go through the commit history [here](https://github.com/fossasia/open-event-next/commits/master/.vscode/settings.json), you will find that you have mistakenly pushed the vscode settings previously. I am trying to remove that and putting it to the `.gitignore` in this pr instead of adding anything. 


Hope your queries got resolved @progmatic-99 , thanks.

File: .gitignore [hunk 7]
```diff
@@ -33,5 +33,8 @@ yarn-error.log*
 # vercel
 .vercel
 
-tsconfig.tsbuildinfo
-.vscode
+# vscode settings
+.vscode/
```
Comment (niloysikdar) [inline .gitignore:37]: @progmatic-99

File: .vscode/settings.json [hunk 8]
```diff
@@ -1 +0,0 @@
-{}
```
Comment (niloysikdar) [inline .vscode/settings.json:1]: @progmatic-99

**Review by niloysikdar — COMMENTED** [review]
Kindly look into these files and changes @progmatic-99

**Review by mariobehling — COMMENTED** [review]
Looks good to me.

**Review by pc-beast — COMMENTED** [review]
Compare
1. https://eventyay.com/e/6b901f56/
2. https://open-event-next-git-fork-niloysikdar-event-page-eventyay.vercel.app/e/6b901f56

| 
image
| 
image |

| 
image
 | 
image
|

There are many other such changes
